decidim-admin/app/commands/decidim/admin/destroy_category.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+module Decidim
+  module Admin
+    # A command with all the business logic to destroy a category in the
+    # system.
+    class DestroyCategory < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # category - A Category that will be destroyed
+      def initialize(category)
+        @category = category
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid.
+      # - :invalid if the data wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if category.nil? || category.subcategories.any?
```

Comment:
Maybe category should known to whom it is related (categorable?) and also block the destruction if there are any.